### PR TITLE
SWE74 Course Form Bugs

### LINKED ISSUE #74: Seat State Not Transferring
## Issue Description
Seat states are not transferring when a new course uses an old template and when a course switches to another course template.

## Steps to Reproduce
- Create a course and change some of the states
then
- Try creating a new course using the previously created template
or
- Create another course then try and using the previously created template

## Tasks
- [x] Find the bug (probably somewhere in the backend)
- [x] Fix the bug

### PR DESCRIPTION
**Pull request by jsmith**
### Changes Proposed in this Pull Request:
- Fixes backend logic for transferring seating templates
- Fixes frontend logic for the disappearing rows / cols
- Fixed jQuery error in course form by moving JavaScript to the bottom (this may be unnecessary now)

### CODE REVIEW

File: src/main/java/com/unb/tracker/web/CourseController.java [hunk 1]
```diff
@@ -201,6 +204,8 @@ private void reuseCourseGridHelper(Course courseReceive, Course courseGive) thro
             newSeat.setRow(seat.getRow());
             newSeat.setCol(seat.getCol());
             newSeat.setState(seat.getState());
+            newSeat.setCourse(courseReceive);
```
Comment (jsmith) [inline src/main/java/com/unb/tracker/web/CourseController.java:239]: This is the line that fixed the backend!

File: src/main/resources/templates/courseFormView.twig [hunk 2]
```diff
@@ -27,11 +27,11 @@
             <input  id="startDate" name="startDate" type="text" class="form-control datepicker" {% if courseCreationError %}value="{{ startDate }}" {% else %} value="{{ course.startDate }}" {% endif %}>
             <label for="startDate">Start Date</label>
         </div>
-        <div class="md-form">
+        <div id="course-form-rows" class="md-form">
```
Comment (jsmith) [inline src/main/resources/templates/courseFormView.twig:32]: This line & the line below it fixed the frontend bug!

File: src/main/resources/templates/course.twig [hunk 3]
```diff
@@ -131,6 +131,30 @@
             username: '{{ user.username }}',
             hasExtendedPrivileges: {{ user.hasExtendedPrivileges }}
         };
+        {% if user.hasExtendedPrivileges %}
+        	$('#confirmDelete').change(function() {
+				$('#deleteButton').attr('disabled',!this.checked);
```
Comment (brandon1024) [inline src/main/resources/templates/course.twig:136]: Theres some pretty crazy tabs going on here!

**Review by brandon1024 — CHANGES_REQUESTED** [review]
Mack and I have noticed some issues on this branch compared to develop. He is getting a 404 (I think). When I try to create a course and use a grid from another course, the course doesn't get created and im just redirected back to the instructor dashboard.

Other than that, these changes look good to me.

**Comment by MackenzieJones** [comment]
When I use seat grid on the dashboard for the first time, it works. But anytime after that, it doesn't. Using seat grid in the editor _works_ every time. No idea why. 
Also, I noticed that the data.sql's course does not have a start date, probably not an issue relating to this though.

EDIT: Alright never mind about the editor seat grid always working. The more I play with it, the more unexpected (broken) results I get.

**Comment by jsmith** [comment]
Ok @brandon1024 , I fixed that issue that you and Mack noticed!

**Review by MackenzieJones — APPROVED** [review]
Tested, looks good

**Comment by brandon1024** [comment]
I will review this morning :)

**Review by brandon1024 — APPROVED** [review]
Looks good. Merging

**Pull request merged**